### PR TITLE
28) added new controllers and connected with it staff

### PR DESCRIPTION
**Pull request by AntonPanchishko**

### CODE REVIEW

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cinema-hall")
```
Comment (less08) [inline src/main/java/com/dev/cinema/controller/CinemaHallController.java]: I beleive mapping should be /cinema-halls

File: src/main/java/com/dev/cinema/controller/MovieController.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie")
```
Comment (less08) [inline src/main/java/com/dev/cinema/controller/MovieController.java]: same, should be movies, plural

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-session")
```
Comment (less08) [inline src/main/java/com/dev/cinema/controller/MovieSessionController.java]: same, plural

File: src/main/java/com/dev/cinema/dao/CinemaHallDao.java [hunk 4]
```diff
@@ -2,9 +2,12 @@
...
 import com.dev.cinema.model.CinemaHall;
 import java.util.List;
+import java.util.Optional;
 
 public interface CinemaHallDao {
     CinemaHall add(CinemaHall cinemaHall);
 
     List<CinemaHall> getAll();
+
+    Optional<CinemaHall> get(Long cinemaHallId);
```
Comment (less08) [inline src/main/java/com/dev/cinema/dao/CinemaHallDao.java]: maybe, it`s better to rename it to getById, so we could understand what exactly is going on

File: src/main/java/com/dev/cinema/dao/MovieDao.java [hunk 5]
```diff
@@ -2,9 +2,12 @@
...
 import com.dev.cinema.model.Movie;
 import java.util.List;
+import java.util.Optional;
 
 public interface MovieDao {
     Movie add(Movie movie);
 
     List<Movie> getAll();
+
+    Optional<Movie> get(Long movieId);
```
Comment (less08) [inline src/main/java/com/dev/cinema/dao/MovieDao.java]: same here

File: src/main/java/com/dev/cinema/dao/MovieSessionDao.java [hunk 6]
```diff
@@ -3,9 +3,16 @@
...
 public interface MovieSessionDao {
     List<MovieSession> findAvailableSessions(Long movieId, LocalDate date);
 
     MovieSession add(MovieSession movieSession);
+
+    MovieSession update(MovieSession movieSession);
+
+    void delete(Long id);
+
+    Optional<MovieSession> get(Long id);
```
Comment (less08) [inline src/main/java/com/dev/cinema/dao/MovieSessionDao.java]: same here

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 7]
```diff
@@ -60,4 +61,58 @@ public MovieSession add(MovieSession movieSession) {
...
+    public void delete(Long id) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            MovieSession movieSession = session.get(MovieSession.class, id);
+            session.delete(movieSession);
+            transaction.commit();
+
```
Comment (less08) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: remove empty line

File: src/main/java/com/dev/cinema/model/dto/response/MovieSessionResponseDto.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
+package com.dev.cinema.model.dto.response;
+
+import com.dev.cinema.model.CinemaHall;
+import com.dev.cinema.model.Movie;
+import java.time.LocalDateTime;
+
+public class MovieSessionResponseDto {
+    private Long id;
+    private Movie movie;
+    private CinemaHall cinemaHall;
```
Comment (less08) [inline src/main/java/com/dev/cinema/model/dto/response/MovieSessionResponseDto.java]: We should not have entity here. Better to pass entity's fields. Like movieTitle

**Review by less08 — CHANGES_REQUESTED** [review]

**Review by less08 — APPROVED** [review]

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**